libraries/WiFi/src/WiFiSTA.h

```diff
@@ -92,6 +92,7 @@ class WiFiSTAClass
     uint8_t subnetCIDR();
     
     bool enableIpV6();
+    bool IPv6(bool state);
     IPv6Address localIPv6();
```

Comment:
Is this now to be marked deprecated (i.e. use IPAddress in preference to IPv6Address).

Also, will the localIP() function be updated to return IPv6 addresses, e.g. if IPv6-only or dual stack?

Reply:
True.

Reply:
True that we don't need `IPv6Address` anymore.

For `localIP()`, we had already the discussion with Core2. For compatibility, I think that `localIP()` should only return IPv4, otherwise a lot of code will break. We need a separate function to get local IPv6

Reply:
Should you be able to guard with `LWIP_IPV6` or maybe some additional attributes?

Legacy code that isn't recompiled should be fine. Legacy code with LWIP_IPV6 will also be fine (previous behaviour).

Even with LWIP_IPV6 turned on, if you don't enable WiFiSTA IPv6 then it won't get a v6 address, i.e. localIP() will only have an IPv4 to return.

At the point you call IPv6(true) to enable IPv6, I don't think it is legacy code anymore, and you shouldn't be surprised if legacy handling of localIP() fails after that.

What would be weird is if you turned on all the IPv6 stuff, e.g. called IPv6(true), and then localIP() did not return IPv6.

Maybe you do need a way to show multiple local addressess, i.e. pick to get only IPv4, but while there might be one IPv4, often there are multiple IPv6.